tests/unit/gtArgsFilter.cpp

```diff
@@ -0,0 +1,392 @@
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+/*
+ * gtArgsFilter.cpp
+ */
+
+#include "argsfilter.h"
+#include "common_test.h"
+#include "gtest/gtest.h"
+using namespace common_test;
+
+class argsfilter : public BaseFixture, public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    m_Properties = NULL;
+    EXPECT_EQ(FPGA_OK, fpgaGetProperties(NULL, &m_Properties));
+  }
+
+  virtual void TearDown() {
+    EXPECT_EQ(FPGA_OK, fpgaDestroyProperties(&m_Properties));
+  }
+
+  fpga_properties m_Properties;
+};
+
+/**
+ * @test       none
+ *
+ * @brief      Given an argument vector with no options related to
+ *             bus, device, function or socket id <br>
+ *             When I call fpgaPropertiesFromArgs with that argument vector<br>
+ *             Then the return value is 0, the fpga_result object is
+ *             unchanged, and my argument vector remains unchanged as well
+ *
+ */
+TEST_F(argsfilter, none) {
+  fpga_result res = FPGA_EXCEPTION;
+  int argc = 5;
+  const char *argv[5] = {"prog", "--opt1", "val1", "word1", "word2"};
+  int exit_code = fpgaPropertiesFromArgs(m_Properties, &res, &argc,
+                                         const_cast<char **>(argv));
+  EXPECT_EQ(exit_code, 0);
+  EXPECT_EQ(res, FPGA_EXCEPTION);
+  EXPECT_EQ(5, argc);
+  EXPECT_STREQ(argv[0], "prog");
+  EXPECT_STREQ(argv[1], "--opt1");
+  EXPECT_STREQ(argv[2], "val1");
+  EXPECT_STREQ(argv[3], "word1");
+  EXPECT_STREQ(argv[4], "word2");
```

Comment:
Not sure that these checks on argc and argv are adding value?

Reply:
This just verifies that calling `fpgaPropertiesFromArgs` (without the relevant arguments) doesn't mutate the argument vector.